h5py/tests/test_errors.py

```diff
@@ -0,0 +1,81 @@
+# This file is part of h5py, a Python interface to the HDF5 library.
+#
+# http://www.h5py.org
+#
+# Copyright 2008-2013 Andrew Collette and contributors
+#
+# License:  Standard 3-clause BSD; see "license.txt" for full license terms
+#           and contributor agreement.
+
+"""
+    Tests the h5py.File object.
+"""
+
+import threading
+import h5py
+
+
+def _access_not_existing_object(filename):
+    """Create a file and access not existing key"""
+    with h5py.File(filename, 'w') as newfile:
+        try:
+            doesnt_exist = newfile['doesnt_exist'].value
+        except KeyError:
+            pass
+
+
+def test_unsilence_errors(tmp_path, capfd):
+    """Chech that HDF5 errors can be mute/unmute from h5py"""
+    filename = str((tmp_path / 'test.h5').resolve())
+
+    # Unmute HDF5 errors
+    h5py._errors.unsilence_errors()
+    _access_not_existing_object(filename)
+    captured = capfd.readouterr()
+    assert captured.err != ''
+    assert captured.out == ''
```

Comment:
As `unsilence_errors` modifies global state, can you wrap everything between *unsilence* and *silence* in a try/finally, so that an error here doesn't leave the state changed for other tests.